scalding-quotation/src/main/scala/com/twitter/scalding/quotation/TreeOps.scala

```diff
@@ -0,0 +1,46 @@
+package com.twitter.scalding.quotation
+
+import scala.reflect.macros.blackbox.Context
+
+trait TreeOps {
+  val c: Context
+  import c.universe._
+
+  /**
+   * Finds the first tree that satisfies the condition.
+   */
+  def find(tree: Tree)(f: Tree => Boolean): Option[Tree] = {
+    var res: Option[Tree] = None
+    val t = new Traverser {
+      override def traverse(t: Tree) = {
+        if (res.isEmpty)
+          if (f(t))
+            res = Some(t)
+          else
+            super.traverse(t)
+      }
+    }
+    t.traverse(tree)
+    res
+  }
+
+  /**
+   * Similar to tree.collect but it doesn't collect the children of a
+   * collected tree.
+   */
+  def collect[T](tree: Tree)(f: PartialFunction[Tree, T]): List[T] = {
+    var res = List[T]()
```

Comment:
what about using a `List.newBuilder[T]` and using `+=` on it which I think gives O(1) append (because they cheat and mutate the list). Currently I think this code is `O(N^2)` since it is using append on a list.

Reply:
fixed